### PR TITLE
Properties to configure hardcoded limits on APIs - JENKINS-34791

### PR DESCRIPTION
**Pull request by svanoort**
Provides java system properties to configure the hardcoded limits on jobs per run, etc per [JENKINS-34791](https://issues.jenkins-ci.org/browse/JENKINS-34791).

### CODE REVIEW

**Comment by svanoort** [comment]
@reviewbybees

**Comment by jglick** [comment]
I recommend filing PRs from forks rather than origin branches.

File: rest-api/src/main/java/com/cloudbees/workflow/rest/external/JobExt.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
     /**
      * Max number of runs per page. Pagination not yet supported.
      */
-    public static final int RUN_PAGE_SIZE = 20;
+    public static final int MAX_RUNS_PER_JOB = Integer.getInteger(JobExt.class.getName()+".maxRunsPerJob", 10);
```
Comment (jglick) [inline rest-api/src/main/java/com/cloudbees/workflow/rest/external/JobExt.java]: Why the reduction?

Comment (jglick) [inline rest-api/src/main/java/com/cloudbees/workflow/rest/external/JobExt.java]: I see, it was already capped in `RunGroupGenerator`.

**Comment by jglick** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by cgomestw** [comment]
have a way to get more runs via curl without change env variable?

**Pull request merged**